CodeEdit/Features/Editor/Views/EditorAreaFileView.swift

```diff
@@ -26,7 +26,7 @@ struct EditorAreaFileView: View {
     @ViewBuilder var editorAreaFileView: some View {
         if let document = file.fileDocument {
 
-            if let utType = document.utType, utType.conforms(to: .text) {
+            if let fileURL = document.fileURL, (try? String(contentsOf: fileURL)) != nil {
```

Comment:
Instead of reading the entire file into memory like this, would it work the same way to just check the document's `content` property for `isEmpty`? That would be a much more efficient operation, as this would potentially read a large file into memory quite often.

Reply:
There is an `!isEmpty` check in the `utType` computed property.

https://github.com/plbstl/CodeEdit/blob/83980b99dec26de57085e7d84901127ea4d1ce78/CodeEdit/Features/Documents/CodeFileDocument.swift#L53-L70

 I attempted different ways to partially read from the file document and determine if it is a text file but I could not get it working well